.github/workflows/docker-dev.yaml

```diff
@@ -0,0 +1,54 @@
+name: Docker Dev
+on:
+  push:
+    branches:
+      - main
+  pull_request_target: {}
+
+# By specifying the access of one of the scopes, all of those that are not specified are set to 'none'.
+permissions:
+  # To be able to access the repository with `actions/checkout`
+  contents: read
+
+jobs:
+  build:
+    name: Build & Push
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Getting image tag
+        id: tag
+        run: |
+          tag="${{ github.sha }}"
+
+          if [[ ${{ github.event.pull_request.head.sha }} != "" ]]; then
+            echo "Detected PR SHA ${{ github.event.pull_request.head.sha }}"
+            tag="${{ github.event.pull_request.head.sha }}"
+          fi
+
+          echo "tag=${tag}" >> $GITHUB_OUTPUT
+          echo "TAG: ${tag}"
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ steps.tag.outputs.tag }}
+          persist-credentials: false
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@ecf95283f03858871ff00b787d79c419715afc34 # v2.7.0
+
+      - name: Login to quay.io
+        uses: docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc # v2.2.0
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_DEV_USERNAME }}
+          password: ${{ secrets.QUAY_DEV_TOKEN }}
+
+      - name: Build & Push Docker Image
+        uses: docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825 # v4.1.1
+        with:
+          context: .
+          push: true
+          platforms: linux/amd64
+          tags: |
+            quay.io/${{ github.repository_owner }}/team-manager-dev:${{ steps.tag.outputs.tag }}
```

Comment:
to be consistent with the remaining workflows, this should be `team-manager-ci` instead